File: frontend/BusinessFlow/src/app/modules/hrm/services/designation.service.ts
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService, PagedResponse } from '../../../core/services/api.service';
import { Designation, DesignationRequest } from '../models/hrm.model';

@Injectable({ providedIn: 'root' })
export class DesignationService {
  private readonly endpoint = '/hr/designations';

  constructor(private api: ApiService) {}

  list(page = 0, size = 20): Observable<PagedResponse<Designation>> {
    return this.api.getPaged<Designation>(this.endpoint, page, size);
  }

  listActive(): Observable<Designation[]> {
    return this.api.get<Designation[]>(`${this.endpoint}/active`);
  }

  create(payload: DesignationRequest): Observable<Designation> {
    return this.api.post<Designation>(this.endpoint, payload);
  }

  update(id: number, payload: DesignationRequest): Observable<Designation> {
    return this.api.put<Designation>(`${this.endpoint}/${id}`, payload);
  }

  toggle(id: number): Observable<Designation> {
    return this.api.patch<Designation>(`${this.endpoint}/${id}/toggle`, {});
  }

  delete(id: number): Observable<string> {
    return this.api.deleteText(`${this.endpoint}/${id}`);
  }
}

File: backend/ZuhooCMS/src/main/java/com/zuhoocms/modules/hrm/designation/DesignationService.java
package com.zuhoocms.modules.hrm.designation;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface DesignationService {

    DesignationResponse create(DesignationRequest request);
    DesignationResponse getById(Long id);
    Page<DesignationResponse> listAll(Pageable pageable);
    List<DesignationResponse> listActive();
    DesignationResponse update(Long id, DesignationRequest request);
    DesignationResponse toggleActive(Long id);
    void delete(Long id);

}
